### PR TITLE
+ RELOP_VARIANT() macro to simplify base_r_relop

### PR DESCRIPTION
**Pull request by romainfrancois**

### CODE REVIEW

**Comment by romainfrancois** [comment]
I believe the errors are unrelated. Merging now.

File: src/relop.cpp [hunk 1]
```diff
@@ -167,55 +167,63 @@ void RelopExecute(DataChunk &args, ExpressionState &state, Vector &result) {
 	RelopExecuteDispatch<LHS_LOGICAL, LHS_TYPE, RHS_LOGICAL, RHS_TYPE, OP>(args, state, result, typename relop_adds_null<LHS_TYPE, RHS_TYPE>::type());
 }
 
+template <LogicalTypeId LOGICAL_TYPE>
+struct physical ;
```
Comment (krlmlr) [inline src/relop.cpp:171]: This might be useful elsewhere, we can move to a header if needed.

**Review by krlmlr — COMMENTED** [review]
Definitely good for avoiding copy-pasties, thanks!

File: src/relop.cpp (hunk 1 above)
Comment (romainfrancois) [reply src/relop.cpp:171]: I agree. Also, I feel it might already exist elsewhere, I'll keep an eye out .

Comment (krlmlr) [reply src/relop.cpp:171]: Not sure. Logical <-> Physical is 1:n in the general case, only for R it's 1:1.

**Pull request merged**